File: app/src/main/java/com/snatik/storage/app/dialogs/RenameDialog.java
package com.snatik.storage.app.dialogs;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.snatik.storage.Storage;
import com.snatik.storage.app.R;

import java.io.File;

/**
 * Created by sromku on June, 2017.
 */
public class RenameDialog extends DialogFragment {

    private final static String PATH = "path";
    private DialogListener mListener;
    private Storage mStorage;

    public static RenameDialog newInstance(String path) {
        RenameDialog fragment = new RenameDialog();
        Bundle args = new Bundle();
        args.putString(PATH, path);
        fragment.setArguments(args);
        return fragment;
    }

    public RenameDialog() {
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        mStorage = new Storage(getActivity());

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        final View view = LayoutInflater.from(getActivity())
                .inflate(R.layout.rename_dialog, (ViewGroup) getView(), false);

        // if text is empty, disable the dialog positive button
        final EditText currentNameText = (EditText) view.findViewById(R.id.current_name);
        String path = getArguments().getString(PATH);

        final File file = mStorage.getFile(path);
        currentNameText.setText(file.getName());
        final String parent = file.getParent();

        final EditText newNameText = (EditText) view.findViewById(R.id.new_name);
        newNameText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void afterTextChanged(Editable editable) {
                ((AlertDialog) getDialog()).getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(editable != null &&
                        editable.length() > 0);
            }
        });

        builder.setTitle(R.string.rename);
        builder.setView(view);
        builder.setPositiveButton(R.string.label_save, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                String newName = newNameText.getText().toString();
                String toPath = parent == null ? newName : parent + File.separator + newName;
                mListener.onRename(file.getPath(), toPath);
            }
        });

        final AlertDialog dialog = builder.create();
        view.post(new Runnable() {
            @Override
            public void run() {
                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false);
            }
        });
        dialog.setCancelable(false);
        return dialog;
    }

    public interface DialogListener {
        void onRename(String fromPath, String toPath);
    }

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        try {
            mListener = (DialogListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement DialogListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

}

File: app/src/main/java/com/snatik/storage/app/dialogs/RenameDialog.kt
package com.snatik.storage.app.dialogs

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.fragment.app.DialogFragment
import com.snatik.storage.Storage
import com.snatik.storage.app.R
import java.io.File

/**
 * Created by sromku on June, 2017.
 */
class RenameDialog : DialogFragment() {
    private var mListener: DialogListener? = null
    private var mStorage: Storage? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        mStorage = Storage(activity!!)

        val builder = AlertDialog.Builder(activity)

        val view = LayoutInflater.from(activity)
                .inflate(R.layout.rename_dialog, view as ViewGroup?, false)

        // if text is empty, disable the dialog positive button
        val currentNameText = view.findViewById<View>(R.id.current_name) as EditText
        val path = arguments?.getString(PATH)

        val file = mStorage!!.getFile(path!!)
        currentNameText.setText(file.name)
        val parent = file.parent

        val newNameText = view.findViewById<View>(R.id.new_name) as EditText
        newNameText.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}

            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}

            override fun afterTextChanged(editable: Editable?) {
                (dialog as AlertDialog).getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = editable != null && editable.length > 0
            }
        })

        builder.setTitle(R.string.rename)
        builder.setView(view)
        builder.setPositiveButton(R.string.label_save) { dialogInterface, i ->
            val newName = newNameText.text.toString()
            val toPath = if (parent == null) newName else parent + File.separator + newName
            mListener!!.onRename(file.path, toPath)
        }

        val dialog = builder.create()
        view.post { dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = false }
        dialog.setCancelable(false)
        return dialog
    }

    interface DialogListener {
        fun onRename(fromPath: String, toPath: String)
    }

    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        try {
            mListener = activity as DialogListener
        } catch (e: ClassCastException) {
            throw ClassCastException("$activity must implement DialogListener")
        }

    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    companion object {

        private val PATH = "path"

        fun newInstance(path: String): RenameDialog {
            val fragment = RenameDialog()
            val args = Bundle()
            args.putString(PATH, path)
            fragment.arguments = args
            return fragment
        }
    }

}
